sbi/inference/base.py

```diff
@@ -152,6 +154,37 @@ def __init__(
             median_observation_distances=[], epochs=[], best_validation_log_probs=[],
         )
 
+    def provide_presimulated(self, theta: Tensor, x: Tensor) -> None:
+        r"""
+        Provide external $\theta$ and $x$ to be used for training later on.
+
+        In this function, $\theta$ and $x$ are merely stored in an attribute. Later on,
+        they will be merged with simulated data, processed (e.g. to get rid of invalid
+        simulations), and eventually put into the banks to be used for training.
+        """
+        self._external_data = (theta, x)
+        self._presimulated_current_round = True
```

Comment:
Why doesn't this function directly call `_append_to_round_bank`?

Reply:
This would in theory be possible. But:
- at this point, the presimulated data is not yet cleaned up (i.e. still contains NaNs)
- we are yet to concatenate the simulated data.

Both of this could be achieved by overwriting the data in `self._theta_bank` later in the code. Imo the iumplemented version is cleaner than that.

Reply:
Regarding the first point, I'd suggest to check for validity inside the `_append` function (see other comments below). Regarding "we are yet to concatenate the simulated data." I can't quite follow -- wouldn't be this done naturally once `_append` is called with simulated data?

Reply:
To elaborate a bit more:

One problem is that, if we access self.theta_bank[0], it has to return the presimulated data _and_ the additional first round simulations. Hence
```
self.theta_bank.append(presimulated)
...
self.theta_bank.append(simulated)
```
does not do the right thing.